guides/keras_tuner/failed_trials.py

```diff
@@ -0,0 +1,221 @@
+unexpected errors. Is there a way to better handle this situation?
+
+We can raise the `FailedTrialError` to skip the retries. Whenever, this error is
+raised, the trial would not be retried. The retries will still run when other
+errors occur. An example is shown as follows.
+"""
+
+
+def build_model(hp):
+    model = keras.Sequential()
+    units_1 = hp.Int("units_1", 10, 40, step=10)
+    units_2 = hp.Int("units_2", 10, 30, step=10)
+
+    # Define the 2 hyperparameters for the units in dense layers
+    model.add(layers.Dense(units=units_1))
+    model.add(layers.Dense(units=units_2))
+
+    model.add(layers.Dense(units=1))
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    if units_1 * units_2 > 800:
+        # When this error is raised, it skips the retries.
+        raise keras_tuner.errors.FailedTrialError("Model too large!")
+    return model
+
+
+tuner = keras_tuner.GridSearch(
+    hypermodel=build_model,
+    objective="val_loss",
+    overwrite=True,
+    max_retries_per_trial=3,
+    max_consecutive_failed_trials=8,
+)
+
+# Use random data to train the model.
+tuner.search(
+    x=np.random.rand(100, 20),
+    y=np.random.rand(100, 1),
+    validation_data=(
+        np.random.rand(100, 20),
+        np.random.rand(100, 1),
+    ),
+    epochs=10,
+)
+
+# Print the results.
+tuner.results_summary()
+
+"""
+## Terminate the search programmatically
+
+When there is a bug in the code we should terminate the search immediately and
+fix the bug. You can terminate the search programmatically when your defined
+conditions are met. Raising a `FatalError` (or its subclasses `FatalValueError`,
+`FatalTypeError`, or `FatalRuntimeError`) will terminate the search regardless
+of the `max_consecutive_failed_trials` argument.
+
+Following is an example to terminate the search when the model is too large.
+"""
+
+
+def build_model(hp):
+    model = keras.Sequential()
+    units_1 = hp.Int("units_1", 10, 40, step=10)
+    units_2 = hp.Int("units_2", 10, 30, step=10)
+
+    # Define the 2 hyperparameters for the units in dense layers
+    model.add(layers.Dense(units=units_1))
+    model.add(layers.Dense(units=units_2))
+
+    model.add(layers.Dense(units=1))
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    if units_1 * units_2 > 800:
+        # When this error is raised, the search is terminated.
+        raise keras_tuner.errors.FatalError("Model too large!")
+    return model
+
+
+tuner = keras_tuner.GridSearch(
+    hypermodel=build_model,
+    objective="val_loss",
+    overwrite=True,
+    max_retries_per_trial=3,
+    max_consecutive_failed_trials=8,
+)
+
+try:
+    # Use random data to train the model.
+    tuner.search(
+        x=np.random.rand(100, 20),
+        y=np.random.rand(100, 1),
+        validation_data=(
+            np.random.rand(100, 20),
+            np.random.rand(100, 1),
+        ),
+        epochs=10,
+    )
+except keras_tuner.errors.FatalError:
```

Comment:
Please add a short "takeaways" section at the end to recapitulate what the reader has learned.

Reply:
Done.